### DISCUSSION TITLE
Hybrid search with dense/binary vectors

### DISCUSSION THREAD

**Discussion by kheyer** [milvus-io/milvus#33381] [Q&A and General discussion]
Is it possible to build/query an index with both dense vectors and binary vectors? For example combining cosine similarity over dense vectors with substructure or jaccard constraint on binary vectors in a single query

**Comment by xiaofan-luan** [comment]
could you give more information.
did you want to do hybrid search? or you want to search with binary vector and refine with float32

**Reply by kheyer** [reply]
Ideally hybrid search like top-k similarity using dense vectors with a constraint on binary vectors. ie top-k dense similarity with a substructure requirement, or top-k dense similarity with tanimoto similarity greater than some value to a reference vector.

**Reply by xiaofan-luan** [reply]
Understand, so create 1 binary vector and 1 float vector, and using RRF or other ranking strategy to fuse the result?

**Comment by xiaofan-luan** [comment]
@czs007 could you help on this?
